# Cinccinnati Paint Company

In [46]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

# Importe de Datos
data = pd.read_csv("./CincinnatiPaintCompany.csv")
data.head()

Commissions  Calls  Driven
0           22    139    2371
1           13    132    2226
2           33    144    2731
3           38    142    3351
4           23    142    2289

# 1. Determinar las variables independientes

In [47]:
X = data[["Driven", "Calls"]]
X 

Driven  Calls
0     2371    139
1     2226    132
2     2731    144
3     3351    142
4     2289    142
5     3449    142
6     3114    138
7     3342    139
8     2842    144
9     2625    134
10    2121    135
11    2219    137
12    3463    146
13    3290    146
14    3103    144
15    2122    147
16    2791    144
17    3209    149
18    2287    131
19    2848    144
20    2690    132
21    2933    132
22    2671    127
23    2988    154
24    2829    147

# #2. Determinar la variable dependiente

In [48]:
y = data[['Commissions']]
y

Commissions
0            22
1            13
2            33
3            38
4            23
5            47
6            29
7            38
8            41
9            32
10           20
11           13
12           47
13           38
14           44
15           29
16           38
17           37
18           14
19           34
20           25
21           27
22           25
23           43
24           34

In [49]:
# Ajustar el modelo de regresión múltiple
model = LinearRegression()
model.fit(X, y) #Entrenando el modelo
y_pred = model.predict(X) #Predicciones del modelo
y_pred

#Métricas estadísticas
mse = mean_squared_error(y, y_pred) # Error de la media cuadrática
r2 = r2_score(y, y_pred) # Varianza

# 3. Valor de Intersección

In [50]:
model.intercept_

array([-101.31444968])

In [51]:
# Coeficientes de la regresión
print("Coeficientes: ", model.coef_)
print("Error de la media cuadrática (MSE): ", mse)
print("Varianza (R^2): ", r2)

Coeficientes:  [[0.01568925 0.63233256]]
Error de la media cuadrática (MSE):  16.313412727936058
Varianza (R^2):  0.8352686374291525


# 4. Estimación del Rendimiento del Trabajador

In [52]:
Y = model.predict(np.array([[2800, 143]]))
print("Rendimiento estimado del trabajador: ", Y)

Rendimiento estimado del trabajador:  [[33.03901808]]


c:\Users\panch\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2749: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Conclusión

Los resultados indican que los agentes que realizan más llamadas de ventas y recorren más millas tienden a obtener mayores comisiones. Ambos coeficientes son positivos, lo que muestra que tanto el esfuerzo en contacto con clientes como el desplazamiento contribuyen al aumento de las comisiones, y el valor de R a la 2 = 0.835, confirma que el modelo explica una proporción muy alta de la variación en los ingresos por comisiones. 